src/StreamSelectLoop.php

```diff
@@ -220,27 +213,117 @@ private function waitForStreamActivity($timeout)
     }
 
     /**
+     * Returns integer amount of seconds in $time or null.
+     *
+     * @param float|null $time – time in seconds
+     *
+     * @return int|null
+     */
+    private static function getSeconds($time)
+    {
+        /*
+         * Workaround for PHP int overflow:
+         * (float)PHP_INT_MAX == PHP_INT_MAX => true
+         * (int)(float)PHP_INT_MAX == PHP_INT_MAX => false
+         * (int)(float)PHP_INT_MAX == PHP_INT_MIN => true
+         */
+        if ($time == PHP_INT_MAX) {
+            return PHP_INT_MAX;
+        }
+
+        return $time === null ? null : intval(floor($time));
```

Comment:
Minor nitpicking: for consistency with the other 2 methods, i'd like to an early return for `$time === null`:

```php
if ($time === null) {
    return null;
}

// ...

return intval(floor($time));
```